src/SprykerSdk/Zed/ComposerConstrainer/Business/Finder/UsedForeignModuleFinder/UsedForeignModuleFinder.php

```diff
@@ -0,0 +1,199 @@
+                $usedModuleCollectionTransfer->addUsedModule($usedModuleTransfer);
+            }
+        }
+
+        return $usedModuleCollectionTransfer;
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return bool
+     */
+    protected function isExcludedClass(string $className): bool
+    {
+        $namespaces = array_merge(
+            $this->config->getCoreNamespaces(),
+            $this->config->getExcludedNameSpaces(),
+            $this->config->getProjectNamespaces()
+        );
+        $pattern = sprintf('/(%s)\\\\/', implode('|', $namespaces));
+
+        return (bool)preg_match_all($pattern, $className);
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\AbstractTransfer\UsedModuleTransfer|null
+     */
+    protected function getUsedModuleByClassName(string $className): ?UsedModuleTransfer
+    {
+        $classFilename = $this->getClassFileNameByClassName($className);
+        $composerJsonData = $this->getComposerJsonDataByClassFilename($classFilename);
+        $packageName = explode('/', $composerJsonData['name']);
+        $usedModuleTransfer = (new UsedModuleTransfer())
+            ->setOrganization($packageName[0])
+            ->setModule($packageName[1]);
+
+        return $usedModuleTransfer;
+    }
+
+    /**
+     * @param string $classFilename
+     *
+     * @return array
+     */
+    protected function getComposerJsonDataByClassFilename(string $classFilename): array
+    {
+        $pattern = sprintf('/(%s)([^\/]+\/+){2}/', str_replace('/', '\/', $this->config->getVendorDirectory()));
+        preg_match($pattern, $classFilename, $matches);
+        $composerJsonFilePath = $matches[0];
+        $composerJsonData = $this->composerJsonReader->readFromFilePath($composerJsonFilePath);
+
+        return $composerJsonData;
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return bool
+     */
+    protected function isVendorClass(string $className): bool
+    {
+        $filename = $this->getClassFileNameByClassName($className);
+        $pattern = sprintf('/%s/', str_replace('/', '\/', $this->config->getVendorDirectory()));
+
+        return (bool)preg_match($pattern, $filename);
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return string
+     */
+    protected function getClassFileNameByClassName(string $className): string
+    {
+        return (new ReflectionClass($className))->getFileName();
+    }
+
+    /**
+     * @param string $fileContent
+     *
+     * @return string[]
+     */
+    protected function getUsedClassesInFile(string $fileContent): array
+    {
+        $pattern = '/^use (.*);/m';
+        $usedClasses = [];
+        if (preg_match_all($pattern, $fileContent, $matches, PREG_SET_ORDER)) {
+            foreach ($matches as $match) {
+                $usedClasses[] = explode(' ', $match[1])[0];
```

Comment:
Would be nice to move this to some additional method e.g. getRealClassName or filterAlias or smth like that.

Reply:
I do believe this wold be redundant as this function is being too coupled to other part of the code. Basically, explode function here does only a small part of the logic - low level technical modification that is too small to give it a standalone name that would fit the meaning of the action itself without the context of the rest of the code.